model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/RoleAdapter.java

```diff
@@ -61,6 +61,15 @@ protected void getDelegateForUpdate() {
         }
     }
 
+    protected void getDelegateForRename(String newName) {
+        if (!Objects.equals(newName, cached.getName())) {
+            // New role name might have been cached as non-existent
+            String containerId = getContainerId();
+            cacheSession.registerRoleInvalidation(cached.getId(), newName, containerId);
+        }
```

Comment:
AFAIK, `newName` is always different. 

https://github.com/keycloak/keycloak/blob/7cff8572387e6bbfca160aa60e9a8191d22e5c85/services/src/main/java/org/keycloak/services/resources/admin/RoleResource.java#L65-L66

I suggest removing the if block. 
```suggestion
        cacheSession.registerRoleInvalidation(cached.getId(), newName, getContainerId());
```

Reply:
OK, it is your call. No objection from me.